shop/src/main/java/ro/msg/learning/shop/configurations/LocationStrategyConfiguration.java

```diff
@@ -0,0 +1,21 @@
+package ro.msg.learning.shop.configurations;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.services.strategy.LocationStrategy;
+import ro.msg.learning.shop.services.strategy.MostAbundantLocationStrategy;
+import ro.msg.learning.shop.services.strategy.SingleLocationStrategy;
+
+@Configuration
+public class LocationStrategyConfiguration {
+
+    @Bean
+    public static LocationStrategy chooseLocationStrategy(@Value("${my.strategy}") String strategy){
+        if(strategy.equals("single")){
```

Comment:
You can create an Enum for Strategy Type and replace the hardcoded text with Enum value